tt_metal/detail/tt_metal.hpp

```diff
@@ -27,7 +27,20 @@ namespace tt::tt_metal{
 
     namespace detail {
 
+        inline bool bypassDispatchCheck = false;
```

Comment:
@pgkeller Because dispatch routing is not enabled during init, we basically can't dispatch programs to remote chips even at the very end of the init process. Profiler time sync process uses a program to sync device and host times. We can just Slow dispatch this program directly from host. These bypasses are needed to allow mixing FD and SD during init.

Reply:
so is the issue that the R chip dispatch path is not set up when R chip profiler is requested to be initialized?

Reply:
Right  `tt::Cluster::instance().set_internal_routing_info_for_ethernet_cores(true); `  is not set yet; when profiler wants to send a sync tt_metal program to the remote chip.

Reply:
But if the R chip has been initialized wouldn't the dispatch path have been set?

Reply:
`tt::Cluster::instance().set_internal_routing_info_for_ethernet_cores(true); ` This is only set after all chips R+L have been initialized. It is a global set, a global switch for all eriscs to enter context switching. @aliuTT Please correct me if I am wrong. So, while, yes they are initialized, they don't see traffic.

Reply:
Yes that's right.